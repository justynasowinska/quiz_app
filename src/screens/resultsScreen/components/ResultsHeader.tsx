import * as React from 'react';
import { Text, StyleSheet } from 'react-native';

import { THEME_COLORS } from '../../../config/colors';

export interface ResultsSummaryType {
    correctAnswers: number;
    allQuestions: number;
}

interface PropsType {
    results: ResultsSummaryType;
}

export const ResultsHeader = (props: PropsType) => {
    const { results: { correctAnswers, allQuestions } } = props;
    return (
        <>
            <Text style={styles.header}>You scored</Text>
            <Text style={[styles.header, styles.scores]}>{`${correctAnswers} / ${allQuestions}`}</Text>
        </>
    );
};

const styles = StyleSheet.create({
    header: {
        fontSize: 22,
        fontWeight: 'bold',
        color: THEME_COLORS.headerTextColor,
        textAlign: 'center'
    },
    scores: {
        fontSize: 24
    }
});