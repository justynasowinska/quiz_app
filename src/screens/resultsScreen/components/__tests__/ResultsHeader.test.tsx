import * as React from 'react';
import { shallow } from 'enzyme';

import { ResultsHeader } from '../ResultsHeader';

describe('Results Screen ResultsHeader Component', () => {
    it('matches to the snapshot', () => {
        const component = shallow(
            <ResultsHeader
                results={{ correctAnswers: 20, allQuestions: 23 }}
            />
        );

        expect(component).toMatchSnapshot();
    });
});
