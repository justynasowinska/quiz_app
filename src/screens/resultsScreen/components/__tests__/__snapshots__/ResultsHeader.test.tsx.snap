// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Screen ResultsHeader Component matches to the snapshot 1`] = `
<Fragment>
  <Text
    style={
      Object {
        "color": "#0C9198",
        "fontSize": 22,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    You scored
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#0C9198",
          "fontSize": 22,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        Object {
          "fontSize": 24,
        },
      ]
    }
  >
    20 / 23
  </Text>
</Fragment>
`;
